### ISSUE TITLE
Parser too flexible for "empty" documents

### ISSUE THREAD

**Issue by swalkinshaw** [rmosolgo/graphql-ruby#2243]
Compared to graphql-js, `GraphQL.parse` seems to be much more flexible and allowing for "empty"-ish documents. Two examples are a query string with comments *only* and just `{}`.

```ruby
[1] pry(main)> GraphQL.parse("#comment")
=> #<GraphQL::Language::Nodes::Document:0x00007ff58dc8ead0 @definitions=[], @filename=nil>
[3] pry(main)> GraphQL.parse("{}")
=> #<GraphQL::Language::Nodes::Document:0x00007ff5813e8f90 @definitions=[#<GraphQL::Language::Nodes::OperationDefinition:0x00007ff5813e91c0 @col=1, @directives=[], @filename=nil, @line=1, @name=nil, @operation_type="query", @selections=[], @variables=[]>], @filename=nil>
```

In graphql-js, both raise an error.

* comments only: `GraphQLError: Syntax Error: Unexpected <EOF>`
* `{}`: `GraphQLError: Syntax Error: Expected Name, found }`

**Comment by rmosolgo** [comment]
Happy to consider a change here, especially if it's in the spec. (And also if it's _not_!)

**Comment by rmosolgo** [comment]
Fixed by #2344
